BLE_PeriodicAdvertising/source/main.cpp

```diff
@@ -445,19 +429,21 @@ class PeriodicDemo : private mbed::NonCopyable<PeriodicDemo>, public ble::Gap::E
     /** Called when a periodic advertising packet is received. */
     virtual void onPeriodicAdvertisingReport(
        const ble::PeriodicAdvertisingReportEvent &event
-    ) {
+    )
+    {
         ble::AdvertisingDataParser adv_parser(event.getPayload());
 
         /* parse the advertising payload, looking for a battery level */
         while (adv_parser.hasNext()) {
             ble::AdvertisingDataParser::element_t field = adv_parser.next();
 
             if (field.type == ble::adv_data_type_t::SERVICE_DATA) {
-                if (memcmp(field.value.data(), _battery_uuid.getBaseUUID(), _battery_uuid.getLen()) != 0) {
+                GattService::UUID_BATTERY_SERVICE;
+                if (*((uint16_t*)field.value.data()) != GattService::UUID_BATTERY_SERVICE) {
                     printf("Unexpected service data\r\n");
                 } else {
                     /* battery level is right after the UUID */
-                    const uint8_t *battery_level = field.value.data() + _battery_uuid.getLen();
+                    const uint8_t *battery_level = field.value.data() + sizeof(uint16_t);
```

Comment:
It's associated with the size of the UUID which may vary, I wouldn't hardcode it even if it is correct in this case, it won't be with 128 bits UUIDS.

Reply:
No, it cannot be 128, adv_data_type_t::SERVICE_DATA defines the payload *must* contain a16bit UUID, if the UUID is 128 bits it will use type SERVICE_DATA_128BIT_ID

Reply:
I forgot about this. Thanks.